test/integration/src/test/java/io/pravega/test/integration/ReadWriteTest.java

```
@@ -0,0 +1,262 @@
+    private Controller controller = null;
+    private int i;
+
+    @Before
+    public void setup() throws Exception {
+
+        final int controllerPort = TestUtils.getAvailableListenPort();
+        final String serviceHost = "localhost";
+        final int servicePort = TestUtils.getAvailableListenPort();
+        final int containerCount = 4;
+
+        try {
+            // 1. Start ZK
+            this.zkTestServer = new TestingServerStarter().start();
+
+            // 2. Start Pravega service.
+            ServiceBuilder serviceBuilder = ServiceBuilder.newInMemoryBuilder(ServiceBuilderConfig.getDefaultConfig());
+            serviceBuilder.initialize();
+            StreamSegmentStore store = serviceBuilder.createStreamSegmentService();
+
+            this.server = new PravegaConnectionListener(false, servicePort, store);
+            this.server.startListening();
+
+            // 3. Start  controller service
+            this.controllerWrapper = new ControllerWrapper(zkTestServer.getConnectString(), false,
+                    controllerPort, serviceHost, servicePort, containerCount);
+            this.controllerWrapper.awaitRunning();
+            this.controller = controllerWrapper.getController();
+        } catch (Exception e) {
+            log.error("Error during setup", e);
+            throw e;
+        }
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            if (this.controllerWrapper != null) {
+                this.controllerWrapper.close();
+                this.controllerWrapper = null;
+            }
+            if (this.server != null) {
+                this.server.close();
+                this.server = null;
+            }
+            if (this.zkTestServer != null) {
+                this.zkTestServer.close();
+                this.zkTestServer = null;
+            }
+        } catch (Exception e) {
+            log.warn("Exception while tearing down", e);
```

Could we rethrow the exception, or avoid catching it, because the log statement shall not appear in debug logs.